src/client.cpp

```diff
@@ -849,8 +908,20 @@ void CClient::OnControllerInMuteMyself ( bool bMute )
     emit ControllerInMuteMyself ( bMute );
 }
 
-void CClient::OnClientIDReceived ( int iChanID )
+void CClient::OnClientIDReceived ( int iServerChanID )
 {
+    // if we have just connected to a running server, iActiveChannels will be 0
+    // if iActiveChannels is not 0, the server must have been restarted on the fly
+    // in that case, channels might have changed, so clear our list to get it afresh.
+    if ( iActiveChannels != 0 )
+    {
+        qInfo() << "> Server restarted?";
```

Comment:
Isn't this better qDebug()?

Reply:
No, I don't think so. The user might want to know that the server restarted, as it may well not be his server. `qDebug()` output is normally suppressed except for debug builds, and even then needs to be specifically enabled. When doing that, it also gets swamped with a lot of output from the Qt graphics code - I tried it!

Reply:
Agreed, this should be `qInfo` as it's something "unexpected in normal operation" but not necessarily a problem (which the user would need warning about).

Reply:
Ok.